dips/dip-8.md

```diff
@@ -66,91 +155,131 @@ A response would look like the following:
 
 ```
 {
-
     "_ObjectType": "CommandRequestObject",
-    "command_type": "FundPullPreApprovalCommand",
-    "cid": "88b282d6181129f682be0421d0ee9887",
-    "command": FundPullPreApprovalCommand(),
+    "command_type": "FundsPullPreApprovalCommand",
+    "cid": "c10db593-ce3d-4ac7-b83d-1e15600e9e63",
+    "command": FundsPullPreApprovalObject(),
 }
 ```
 
-### FundPullPreApprovalCommand object
-| Field 	    | Type 	| Required? 	| Description 	|
-|-------	    |------	|-----------	|-------------	|
-| _ObjectType   | str  | Y             | The fixed string `FundPullPreApprovalCommand` |
-| fund_pull_pre_approval| [`FundPullPreApprovalObject`](#fundpullpreapprovalobject) | Y | contains a `FundPullPreApprovalObject` that either creates a new pre-approval or updates an existing pre-approval. Note that strict validity checks apply when updating pre-approvals, that are listed in the section below describing these Objects. An invalid update or initial pre-approval object results in a Command error
-| _reads | JSON Object map |  Y | Must be an Object containing mappings of `reference_id` to latest Version as represented by the `cid` of the latest Command which successfully mutated the Object referenced by the `reference_id`. The value in this field must match a `cid` previously specified by a Command by the `_writes` parameter on a prior Command for this payment Object.  For a fund_pull_pre_approval Command, only zero or one `_reads` values should be specified since fund approvals are only dependent upon at most one prior Version of an Object. A list with any other number of items results in a Command error.  If the list is empty this fund_pull_pre_approval Command defines a new fund_pull_pre_approval. If the list contains one item, then this Command updates the shared [`FundPullPreApprovalObject`](#fundpullpreapprovalobject). |
-| _writes | JSON object map | Y | For a fund_pull_pre_approval Command, the `_writes` field may only be a single key-value pair since a fund_pull_pre_approval Command only operates upon one Object.  This field maps the `reference_id` of the Object being written to its new Version. |
+### FundsPullPreApprovalObject
+
+The structure in this object must be a full funds pull pre-approval and all fields beside the `status` are immutable after the creation. In order to change the status, one may send only the `funds_pull_pre_approval_id` and `status` fields. Updating immutable fields with different values results in a command error, but it is acceptable to re-send the same value.
+It should be noted that initial creation of the FundsPullPreApprovalObject can be done by the VASP on either side (buyer or merchant). 
+
+
+#### ___The buyer creates the initial request___
+A merchant doesn't knows the buyer details upfront, so it use an out-of-band channel to request a consent (e.g. QR code). The buyer wallet (VASP) decode the request and initiate FundsPullPreApproval flow from its side
```

Comment:
Just repeating the buyer/merchant link vs payment sender / payment receiver used in DIP 158. Would love consistency.

Reply:
When working on DIP-8 we will use the same terminology.
Resolved for now.